test/e2e/certificates_test.go

```diff
@@ -0,0 +1,102 @@
+// able to use it.
+func (tc *testContext) testKubeletCARotation(t *testing.T) {
+	// force CA rotation
+	err := tc.forceKubeApiServerCertificateRotation()
+	require.NoError(t, err)
+
+	// wait 5 seconds for the new certificate and add 5 seconds per node to allow
+	// WMCO to update the kubelet-ca.crt file on each node
+	nodeCount := len(gc.allNodes())
+	time.Sleep(time.Duration(5+5*nodeCount) * time.Second)
+
+	// get the ConfigMap that contains the kubelet client CA
+	cm, err := tc.client.K8s.CoreV1().ConfigMaps(certificates.KubeApiServerOperatorNamespace).Get(context.TODO(),
+		certificates.KubeAPIServerServingCAConfigMapName, meta.GetOptions{})
+	if err != nil {
+		require.NoError(t, err, "error getting kubelet client CA ConfigMap")
+	}
+	// parse bundle from ConfigMap
+	bundleCABytes, err := certificates.GetCAsFromConfigMap(cm, certificates.CABundleKey)
+	if err != nil {
+		require.NoError(t, err, "error parsing CA bundle from ConfigMap")
+	}
+	bundleCAString := strings.TrimSpace(string(bundleCABytes))
+	// loop all Windows nodes
+	for _, winNode := range gc.allNodes() {
+		// spin-off subtests to ensure the CA bundle file is verified in all Windows nodes
+		t.Run("node/"+winNode.Name, func(t *testing.T) {
+			found, err := tc.isCertificateInCABundle(&winNode, bundleCAString)
+			require.NoError(t, err, "CA bundle file must be available")
+			assert.True(t, found, "certificate should be present in CA bundle")
+		})
+	}
+}
+
+// forceKubeApiServerCertificateRotation forces the certificate rotation by setting the `certificate-not-after`
+// annotation to nil in the `kube-apiserver-to-kubelet-signer` secret
+func (tc *testContext) forceKubeApiServerCertificateRotation() error {
+	// note that the slash had to be encoded as ~1 because it's a reference: https://www.rfc-editor.org/rfc/rfc6901#section-3
+	patchData := fmt.Sprint(`[{"op":"replace","path":"/metadata/annotations/auth.openshift.io~1certificate-not-after","value": null }]`)
+	// invoke request
+	_, err := tc.client.K8s.CoreV1().Secrets("openshift-kube-apiserver-operator").Patch(context.TODO(),
+		"kube-apiserver-to-kubelet-signer", types.JSONPatchType, []byte(patchData), meta.PatchOptions{})
+	return err
+}
+
+// isCertificateInCABundle returns True when the CA bundle file in the given Windows node contains certificate content,
+// False otherwise.
+func (tc *testContext) isCertificateInCABundle(node *core.Node, content string) (bool, error) {
+	addr, err := controllers.GetAddress(node.Status.Addresses)
+	if err != nil {
+		return false, err
+	}
+	// CA bundle location in Windows node. i.e. windows.GetK8sDir() + nodeconfig.KubeletClientCAFilename
+	caBundlePath := "C:\\k\\kubelet-ca.crt"
+	// PowerShell command to find the given content in the file
+	command := fmt.Sprintf("Get-Content -Raw -Path %s | Select-String -SimpleMatch -Quiet -Pattern \\\"%s\\\"",
+		caBundlePath, content)
```

Comment:
I'd prefer we make these powershell commands as simple as possible, as they are harder to tweak and maintain.
Lets just grab the file contents using powershell and do the string comparison here in go.

Reply:
Indeed, I had brought the comparison to this execution thread already as a walkaround for the PowerShell issues I was facing.